File: python/src/falcon_core/instrument_interfaces/port_transforms/port_transforms.py
from __future__ import annotations
from typing import Any, List, Dict, Tuple, Optional
from falcon_core._capi.port_transforms import PortTransforms as _CPortTransforms
from falcon_core.generic.list import List
from falcon_core.instrument_interfaces.port_transforms.port_transform import PortTransform

class PortTransforms:
    """Python wrapper for PortTransforms."""

    def __init__(self, c_obj):
        self._c = c_obj

    @classmethod
    def _from_capi(cls, c_obj):
        if c_obj is None:
            return None
        return cls(c_obj)

    @classmethod
    def PortTransforms_create_empty(cls, ) -> PortTransforms:
        return cls(_CPortTransforms.PortTransforms_create_empty())

    @classmethod
    def PortTransforms_create_raw(cls, data: PortTransform, count: Any) -> PortTransforms:
        return cls(_CPortTransforms.PortTransforms_create_raw(data._c, count))

    @classmethod
    def PortTransforms_create(cls, handle: List) -> PortTransforms:
        return cls(_CPortTransforms.PortTransforms_create(handle._c))

    @classmethod
    def PortTransforms_from_json_string(cls, json: str) -> PortTransforms:
        return cls(_CPortTransforms.PortTransforms_from_json_string(json))

    def transforms(self, ) -> List:
        ret = self._c.transforms()
        if ret is None: return None
        return List(ret)

    def push_back(self, value: PortTransform) -> None:
        ret = self._c.push_back(value._c)
        return ret

    def size(self, ) -> None:
        ret = self._c.size()
        return ret

    def empty(self, ) -> None:
        ret = self._c.empty()
        return ret

    def erase_at(self, idx: Any) -> None:
        ret = self._c.erase_at(idx)
        return ret

    def clear(self, ) -> None:
        ret = self._c.clear()
        return ret

    def at(self, idx: Any) -> PortTransform:
        ret = self._c.at(idx)
        if ret is None: return None
        return PortTransform._from_capi(ret)

    def items(self, ) -> List:
        ret = self._c.items()
        if ret is None: return None
        return List(ret)

    def contains(self, value: PortTransform) -> None:
        ret = self._c.contains(value._c)
        return ret

    def index(self, value: PortTransform) -> None:
        ret = self._c.index(value._c)
        return ret

    def intersection(self, other: PortTransforms) -> PortTransforms:
        ret = self._c.intersection(other._c)
        return cls._from_capi(ret)

    def equal(self, b: PortTransforms) -> None:
        ret = self._c.equal(b._c)
        return ret

    def not_equal(self, b: PortTransforms) -> None:
        ret = self._c.not_equal(b._c)
        return ret

    def to_json_string(self, ) -> str:
        ret = self._c.to_json_string()
        return ret

    def __eq__(self, other):
        """Operator overload for =="""
        if not isinstance(other, PortTransforms):
            return NotImplemented
        return self.equality(other)

    def __ne__(self, other):
        """Operator overload for !="""
        if not isinstance(other, PortTransforms):
            return NotImplemented
        return self.notequality(other)

File: ocaml/_opam/.opam-switch/sources/falcon_core/src/falcon_core/instrument_interfaces/port_transforms/porttransforms.ml
open Ctypes
open Capi_bindings
open ErrorHandling

open Falcon_core.Generic
open Falcon_core.Instrument_interfaces.Port_transforms

class c_porttransforms (h : unit ptr) = object(self)
  val raw_val = h
  method raw = raw_val
  initializer Gc.finalise (fun _ ->
    Capi_bindings.porttransforms_destroy raw_val;
    ErrorHandling.raise_if_error ()
  ) self
end

module PortTransforms = struct
  type t = c_porttransforms

  let copy (handle : t) : t =
    ErrorHandling.read handle (fun () ->
      let ptr = Capi_bindings.porttransforms_copy handle#raw in
      ErrorHandling.raise_if_error ();
      new c_porttransforms ptr
    )

  let fromjson (json : string) : t =
    let ptr = Capi_bindings.porttransforms_from_json_string (Capi_bindings.string_wrap json) in
    ErrorHandling.raise_if_error ();
    new c_porttransforms ptr

  let empty () : t =
    let ptr = Capi_bindings.porttransforms_create_empty () in
    ErrorHandling.raise_if_error ();
    new c_porttransforms ptr

  let make (handle : ListPortTransform.t) : t =
    ErrorHandling.read handle (fun () ->
      let ptr = Capi_bindings.porttransforms_create handle#raw in
      ErrorHandling.raise_if_error ();
      new c_porttransforms ptr
    )

  let equal (handle : t) (other : t) : bool =
    ErrorHandling.multi_read [handle; other] (fun () ->
      let result = Capi_bindings.porttransforms_equal handle#raw other#raw in
      ErrorHandling.raise_if_error ();
      result
    )

  let notEqual (handle : t) (other : t) : bool =
    ErrorHandling.multi_read [handle; other] (fun () ->
      let result = Capi_bindings.porttransforms_not_equal handle#raw other#raw in
      ErrorHandling.raise_if_error ();
      result
    )

  let toJsonString (handle : t) : string =
    ErrorHandling.read handle (fun () ->
      let result = Capi_bindings.porttransforms_to_json_string handle#raw in
      ErrorHandling.raise_if_error ();
      Capi_bindings.string_to_ocaml result
    )

  let transforms (handle : t) : ListPortTransform.t =
    ErrorHandling.read handle (fun () ->
      let result = Capi_bindings.porttransforms_transforms handle#raw in
      ErrorHandling.raise_if_error ();
      new c_listporttransform result
    )

  let pushBack (handle : t) (value : PortTransform.t) : unit =
    ErrorHandling.multi_read [handle; value] (fun () ->
      let result = Capi_bindings.porttransforms_push_back handle#raw value#raw in
      ErrorHandling.raise_if_error ();
      result
    )

  let size (handle : t) : int =
    ErrorHandling.read handle (fun () ->
      let result = Capi_bindings.porttransforms_size handle#raw in
      ErrorHandling.raise_if_error ();
      result
    )

  let empty (handle : t) : bool =
    ErrorHandling.read handle (fun () ->
      let result = Capi_bindings.porttransforms_empty handle#raw in
      ErrorHandling.raise_if_error ();
      result
    )

  let eraseAt (handle : t) (idx : int) : unit =
    ErrorHandling.read handle (fun () ->
      let result = Capi_bindings.porttransforms_erase_at handle#raw (Unsigned.Size_t.of_int idx) in
      ErrorHandling.raise_if_error ();
      result
    )

  let clear (handle : t) : unit =
    ErrorHandling.read handle (fun () ->
      let result = Capi_bindings.porttransforms_clear handle#raw in
      ErrorHandling.raise_if_error ();
      result
    )

  let at (handle : t) (idx : int) : PortTransform.t =
    ErrorHandling.read handle (fun () ->
      let result = Capi_bindings.porttransforms_at handle#raw (Unsigned.Size_t.of_int idx) in
      ErrorHandling.raise_if_error ();
      new c_porttransform result
    )

  let items (handle : t) : ListPortTransform.t =
    ErrorHandling.read handle (fun () ->
      let result = Capi_bindings.porttransforms_items handle#raw in
      ErrorHandling.raise_if_error ();
      new c_listporttransform result
    )

  let contains (handle : t) (value : PortTransform.t) : bool =
    ErrorHandling.multi_read [handle; value] (fun () ->
      let result = Capi_bindings.porttransforms_contains handle#raw value#raw in
      ErrorHandling.raise_if_error ();
      result
    )

  let index (handle : t) (value : PortTransform.t) : int =
    ErrorHandling.multi_read [handle; value] (fun () ->
      let result = Capi_bindings.porttransforms_index handle#raw value#raw in
      ErrorHandling.raise_if_error ();
      result
    )

  let intersection (handle : t) (other : t) : t =
    ErrorHandling.multi_read [handle; other] (fun () ->
      let result = Capi_bindings.porttransforms_intersection handle#raw other#raw in
      ErrorHandling.raise_if_error ();
      new c_porttransforms result
    )

end